src/config/containers-and-vms/chroot.md

```diff
@@ -0,0 +1,90 @@
+# Creating and using chroots and containers
+
+chroots and containers can be set up and used for many purposes, including:
+
+- running glibc software on musl (and vice versa)
+- running software in a more controlled or sandboxed environment
+- creating a rootfs for bootstrapping a system
+
+## Chroot Creation
+
+Create a directory that will contain the chroot, then install a base system in
+it via the `base-voidstrap` package:
+
+```
+$ XBPS_TARGET_ARCH=<chroot_arch> xbps-install -M -r <chroot_dir> -R <repository> base-voidstrap
+```
+
+The `<repository>` varies depending on architecture:
+
+- `aarch64`, `aarch64-musl`:
+   `https://repo-default.voidlinux.org/current/aarch64`
+- any other `musl` architecture:
+   `https://repo-default.voidlinux.org/current/musl`
+- any other architecture: `https://repo-default.voidlinux.org/current`
+
+Alternatively, this can be done automatically with
+[`xvoidstrap(1)`](https://man.voidlinux.org/xvoidstrap.1) (from `xtools`):
+
+```
+# XBPS_ARCH=<chroot_arch> xvoidstrap <chroot_dir> base-voidstrap
+```
+
+## Chroot Usage
+
+### xchroot
+
+[`xchroot(1)`](https://man.voidlinux.org/xchroot.1) (from `xtools`) can be used
+to automatically set up and enter the chroot.
+
+### Manual Method
+
+Alternatively, this process can be done manually.
+
+If network access is required, copy `/etc/resolv.conf` into the chroot;
+`/etc/hosts` may need to be copied as well.
+
+Several directories then need to be mounted as follows:
+
+```
+# mount -t proc none <chroot_dir>/proc
+# mount -t sysfs none <chroot_dir>/sys
+# mount --rbind /dev <chroot_dir>/dev
+# mount --rbind /run <chroot_dir>/run
+```
+
+Use [chroot(1)](https://man.voidlinux.org/chroot.1) to change to the new root,
+then run programs and do tasks as usual. Once you've finished using it, unmount
+the chroot using [umount(8)](https://man.voidlinux.org/umount.8).
+
+### Alternatives
+
+#### Bubblewrap
+
+[bwrap(1)](https://man.voidlinux.org/bwrap.1) (from the `bubblewrap` package)
+has additional features like the ability for sandboxing and does not require
+root access.
+
+`bwrap` is very flexible and can be used in many ways, for example:
+
+```
+$ bwrap --bind <chroot_dir> / \
+	--dev /dev \
+	--proc /proc \
+	--bind /sys /sys \
+	--bind /run /run \
+	--ro-bind /etc/resolv.conf /etc/resolv.conf \
+	<command>
+```
+
+The `bwrap` manpage and the [Arch Wiki
+article](https://wiki.archlinux.org/title/Bubblewrap#Usage_examples) contain
+more examples of `bwrap` usage.
+
+#### Application Containers
```

Comment:
I wonder if this would be also a good place to mention flatpak and link its page. It is a great way to run graphical applications in a "glibc container" and many people don't know this.